### ISSUE TITLE
Crash in Windows 10 in filesystem mode on one project.

### ISSUE THREAD

**Issue by derac** [trufflesecurity/trufflehog#287]
**Describe the bug**
Ran Trufflehog in filesystem mode on a git project and it crashed. It worked fine on a dozen or so others that I've tried.

**To Reproduce**
1. run Trufflehog in filesystem mode on a git directory which I cannot share the details of, sorry.

**Expected behavior**
no crash

**Output**
Before this error, it found four secrets and validated them.
```
2022/04/04 19:17:17 [updater parent] checking for updates...
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x28 pc=0x1a7a4a9]

goroutine 36 [running]:
github.com/trufflesecurity/trufflehog/v3/pkg/sources/filesystem.(*Source).Chunks.func2({0xc00074cc18?, 0x17?}, {0xc0013c7c08?, 0xddf4f8?}, {0x0?, 0x0?})
        F:/Projects/trufflehog/pkg/sources/filesystem/filesystem.go:100 +0xe9
io/fs.walkDir({0x26acde0, 0xc0006d2020}, {0xc00074cc18, 0x17}, {0x26ba980, 0xc000dff340}, 0xc0013c7f50)
        G:/Program Files/Go/src/io/fs/walk.go:66 +0x78
io/fs.walkDir({0x26acde0, 0xc0006d2020}, {0xc00074cb58, 0x11}, {0x26ba980, 0xc000dd2e00}, 0xc0013c7f50)
        G:/Program Files/Go/src/io/fs/walk.go:85 +0x27d
io/fs.walkDir({0x26acde0, 0xc0006d2020}, {0xc0007162f0, 0xc}, {0x26ba980, 0xc000d683f0}, 0xc0013c7f50)
        G:/Program Files/Go/src/io/fs/walk.go:85 +0x27d
io/fs.walkDir({0x26acde0, 0xc0006d2020}, {0x22e4a20, 0x1}, {0x26ba3d0, 0xc0006d2030}, 0xc0013c7f50)
        G:/Program Files/Go/src/io/fs/walk.go:85 +0x27d
io/fs.WalkDir({0x26acde0, 0xc0006d2020}, {0x22e4a20, 0x1}, 0xc000ebdf50)
        G:/Program Files/Go/src/io/fs/walk.go:112 +0xde
github.com/trufflesecurity/trufflehog/v3/pkg/sources/filesystem.(*Source).Chunks(0xc000d341b0, {0x26b9aa0?, 0xc000050048}, 0xc0005d4360)
        F:/Projects/trufflehog/pkg/sources/filesystem/filesystem.go:92 +0x278
github.com/trufflesecurity/trufflehog/v3/pkg/engine.(*Engine).ScanFileSystem.func1()
        F:/Projects/trufflehog/pkg/engine/filesystem.go:31 +0x36
created by github.com/trufflesecurity/trufflehog/v3/pkg/engine.(*Engine).ScanFileSystem
        F:/Projects/trufflehog/pkg/engine/filesystem.go:30 +0x29b
2022/04/04 19:17:18 [updater parent] prog exited with 2
```

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Version: 10.0.19044 Build 19044

**Additional info**
~~I tried git mode on the local directory and it didn't find secrets, just crashed. I am going to make a separate issue for this as it just crashes for all git dirs in this way. git mode works fine on remote dirs.~~ EDIT: I figured out that `file:///F:/` crashes while `file://F:/` does not. The former is valid afaik: https://en.wikipedia.org/wiki/File_URI_scheme

Also, when I run in git mode against this same directory, no secrets are found and trufflehog exits successfully. The tools runs successfully even if the directory doesn't exist, which seems to be why this is happening.

This is a JS project with a node_modules dir btw. Perhaps this has something to do with Windows symlinks?

**Comment by dustin-decker** [comment]
I think the filesystem scanner expects regular paths - not a file URI scheme. 
There's definitely a panic issue to look into here though.

**Comment by derac** [comment]
The direct path gives the error below. Also tried `\\` and `\ `

```
PS F:\Projects\trufflehog> trufflehog git "f:/Projects/REDACTED" --debug
time="2022-04-04T23:47:42-05:00" level=debug msg="running version dev"
2022/04/04 23:47:42 [updater parent] run
2022/04/04 23:47:42 [updater parent] checking for updates...
time="2022-04-04T23:47:43-05:00" level=debug msg="fetching trufflehog update"
2022/04/04 23:47:43 [updater parent] failed to get latest version: Failed to read zip archive: zip: not a valid zip file
2022/04/04 23:47:43 [updater parent] starting C:\Users\Derek\go\bin\trufflehog.exe
time="2022-04-04T23:47:44-05:00" level=debug msg="running version dev"
2022/04/04 23:47:44 [updater child#1] run
2022/04/04 23:47:44 [updater child#1] start program
trufflehog dev
time="2022-04-04T23:47:44-05:00" level=debug msg="running with up to 8 workers"
time="2022-04-04T23:47:44-05:00" level=debug msg="loaded 1 decoders"
time="2022-04-04T23:47:44-05:00" level=debug msg="loaded 610 detectors total, 610 with verification enabled. 0 with verification disabled"
time="2022-04-04T23:47:44-05:00" level=info msg="starting pprof and fgprof server on :18066 /debug/pprof and /debug/fgprof"
time="2022-04-04T23:47:44-05:00" level=fatal msg="error preparing git repo for scanning" error="unsupported Git URI: f:/Projects/REDACTED"
2022/04/04 23:47:44 [updater parent] prog exited with 1
```

**Comment by bill-rich** [comment]
For `git`, the URI will need to start with `file://` for scanning a local repo. When you're doing a `filesystem` scan (which can be pointed at any directory on a filesystem while `git` will only work with a valid git repo), you'll you just need to provide a path. For example, `git filesystem --directory /tmp` will scan the `/tmp` direcectory.

I've got a PR open to fix the crash and provide better logging.

**Comment by dustin-decker** [comment]
Fixed in v3.0.3 (building now)

**Comment by dustin-decker** [comment]
Actually fixed by v3.0.4 (building now) in #299
